packages/eas-cli/src/analytics.ts

```diff
@@ -63,41 +88,73 @@ export async function setUserDataAsync(
 }
 
 export async function flushAsync(): Promise<void> {
-  if (client) {
-    await client.flush();
+  if (rudderstackClient) {
+    rudderstackClient.flush();
+  }
+
+  if (amplitudeClient) {
+    await amplitudeClient.flush();
   }
 }
 
 export function logEvent(name: string, properties: Record<string, any> = {}) {
-  if (client) {
-    ensureUserIdentified();
+  if (!amplitudeClient && !rudderstackClient) {
+    return;
+  }
+  ensureUserIdentified();
+
+  if (amplitudeClient) {
     const { userId, deviceId } = identifyData ?? {};
-    client.logEvent({
+    amplitudeClient.logEvent({
       event_type: name,
-      event_properties: properties,
+      event_properties: { ...properties, source_version: packageJSON?.version, source: 'eas cli' },
       ...(userId && { user_id: userId }),
       ...(deviceId && { device_id: deviceId }),
-      ...getContext(),
+      ...getAmplitudeContext(),
+    });
+  }
+
+  if (rudderstackClient) {
+    const { userId, deviceId } = identifyData ?? {};
+    const identity = { userId: userId ?? undefined, anonymousId: deviceId ?? uuidv4() };
+    rudderstackClient.track({
+      event: name,
+      properties: { ...properties, source_version: packageJSON?.version, source: 'eas cli' },
+      ...identity,
+      context: getRudderStackContext(),
     });
   }
 }
 
 function ensureUserIdentified() {
-  if (client && !userIdentifyCalled && identifyData) {
-    client.identify(identifyData.userId, identifyData.deviceId, identifyData.identify);
-    userIdentifyCalled = true;
+  if (!(rudderstackClient && amplitudeClient) || userIdentified || !identifyData) {
+    return;
   }
+
+  amplitudeClient.identify(identifyData.userId, identifyData.deviceId, identifyData.identify);
+  rudderstackClient.identify({
+    userId: identifyData.userId,
+    anonymousId: identifyData.deviceId,
+  });
```

Comment:
```suggestion
  if (!(rudderstackClient || amplitudeClient) || userIdentified || !identifyData) {
    return;
  }

  if (amplitudeClient) {
    amplitudeClient.identify(identifyData.userId, identifyData.deviceId, identifyData.identify);
  }
  
  if (rudderstackClient) {
    rudderstackClient.identify({
      userId: identifyData.userId,
      anonymousId: identifyData.deviceId,
    });
  }
```

I think this would match the logic in other functions better.